core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp

```diff
@@ -0,0 +1,53 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+template <class T>
+void test_empty_view_runtime_unmanaged() {
+  T d{};
+  auto* p = reinterpret_cast<T*>(0xABADBABE);
+
+  Kokkos::View<T> u0(p);
+  Kokkos::View<T> v0(&d);
+  Kokkos::View<T> w0(nullptr);
+  Kokkos::View<T> x0(0);
+  Kokkos::View<T> y0(NULL);
+
+  Kokkos::View<T*> u1(p, 0);
+  Kokkos::View<T*> v1(&d, 0);
+  Kokkos::View<T*> w1(nullptr, 0);
+  Kokkos::View<T*> x1(0, 0);
+  Kokkos::View<T*> y1(NULL, 0);
+
+  Kokkos::View<T**> u2(p, 0, 0);
+  Kokkos::View<T**> v2(&d, 0, 0);
+  Kokkos::View<T**> w2(nullptr, 0, 0);
+  Kokkos::View<T**> x2(0, 0, 0);
+  Kokkos::View<T**> y2(NULL, 0, 0);
```

Comment:
Don't you get unused-type warnings for all of these?

Reply:
Oversight